peachjam/js/components/DocumentContent/index.ts

```diff
@@ -147,16 +153,38 @@ class DocumentContent {
     , 200));
   }
 
+  setupBookTocActivity () {
+    this.tocController?.addEventListener('itemTitleClicked', (e) => {
+      const customEvt = e as CustomEvent;
+      const id = customEvt.detail.target.getAttribute('href').replace('#', '');
+      const documentElement = this.root.querySelector('[data-document-element]');
+      if (!documentElement) return;
+      if (!(this.storedDocument && this.storedDocument instanceof HTMLElement)) return;
+      if (id) {
+        const sectionOfFocus = this.storedDocument.querySelector(`#${id}`)?.cloneNode(true) as HTMLElement | undefined;
+        if (!sectionOfFocus) return;
+        // Delete content within document element and then append section of focus
+        documentElement.replaceChildren(sectionOfFocus);
+      } else {
+        const storedDocument = this.storedDocument.cloneNode(true);
+        // @ts-ignore
+        documentElement.replaceChildren(...this.storedDocument.children);
+        this.storedDocument = storedDocument;
+      }
+      this.enchrichmentsManager?.layoutItems();
```

Comment:
Is this required? The gutter watches the main container and reflows when its children change, which should be sufficient.

Reply:
good catch, nope this is not required. refactored